api/indexer/client_test.go

```diff
@@ -0,0 +1,182 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package indexer
+
+import (
+	"context"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/trace"
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/hypersdk/api"
+	"github.com/ava-labs/hypersdk/chain/chaintest"
+)
+
+func TestIndexerClientBlocks(t *testing.T) {
+	const (
+		numExecutedBlocks = 4
+		blockWindow       = 2
+		numTxs            = 3
+	)
+
+	testCases := []struct {
+		name      string
+		blkHeight uint64
+		err       error
+	}{
+		{
+			name:      "success",
+			blkHeight: numExecutedBlocks - 1,
+			err:       nil,
+		},
+		{
+			name:      "missing block",
+			blkHeight: 0,
+			err:       errBlockNotFound,
+		},
+	}
+
+	for _, tt := range testCases {
+		t.Run(tt.name, func(t *testing.T) {
+			require := require.New(t)
+			ctx := context.Background()
+			indexer, executedBlocks, _ := createTestIndexer(t, ctx, numExecutedBlocks, blockWindow, numTxs)
+
+			jsonHandler, err := api.NewJSONRPCHandler(Name, NewServer(trace.Noop, indexer))
+			require.NoError(err)
+
+			httpServer := httptest.NewServer(jsonHandler)
+			t.Cleanup(func() {
+				httpServer.Close()
+			})
+
+			parser := chaintest.NewTestParser()
+
+			client := NewClient(httpServer.URL)
+			executedBlock, err := client.GetBlockByHeight(ctx, executedBlocks[tt.blkHeight].Block.Hght, parser)
+			if tt.err == nil {
+				require.NoError(err)
+				require.Equal(executedBlocks[tt.blkHeight].Block, executedBlock.Block)
+			} else {
+				require.Contains(err.Error(), tt.err.Error())
+			}
+
+			executedBlock, err = client.GetBlock(ctx, executedBlocks[tt.blkHeight].Block.GetID(), parser)
+			if tt.err == nil {
+				require.NoError(err)
+				require.Equal(executedBlocks[tt.blkHeight].Block, executedBlock.Block)
+			} else {
+				require.Contains(err.Error(), tt.err.Error())
+			}
+
+			executedBlock, err = client.GetLatestBlock(ctx, parser)
+			require.NoError(err)
+			require.Equal(executedBlocks[numExecutedBlocks-1].Block, executedBlock.Block)
+		})
+	}
+}
+
+func TestIndexerClientTransactions(t *testing.T) {
+	const (
+		numExecutedBlocks = 4
+		blockWindow       = 2
+		numTxs            = 3
+	)
+
+	testCases := []struct {
+		name     string
+		txBlkIdx int
+		txIdx    int
```

Comment:
Can we use `blockIndex` and `txIndex` for consistency? (we never use `Idx` for index in the codebase)